src/analysis/normed_space/pointwise.lean

```diff
@@ -95,13 +95,110 @@ by rw [vadd_ball, vadd_eq_add, add_zero]
 lemma vadd_closed_ball_zero (x : E) (r : ℝ) : x +ᵥ closed_ball 0 r = closed_ball x r :=
 by rw [vadd_closed_ball, vadd_eq_add, add_zero]
 
-variables [normed_space ℝ E]
+variables [normed_space ℝ E] {x y z : E} {δ ε : ℝ}
 
 /-- In a real normed space, the image of the unit ball under scalar multiplication by a positive
 constant `r` is the ball of radius `r`. -/
 lemma smul_unit_ball_of_pos {r : ℝ} (hr : 0 < r) : r • ball 0 1 = ball (0 : E) r :=
 by rw [smul_unit_ball hr.ne', real.norm_of_nonneg hr.le]
 
+-- This is also true for `ℚ`-normed spaces
+lemma exists_dist_eq (x z : E) {a b : ℝ} (ha : 0 ≤ a) (hb : 0 ≤ b) (hab : a + b = 1) :
+  ∃ y, dist x y = b * dist x z ∧ dist y z = a * dist x z :=
+begin
+  refine ⟨a • x + b • z, _, _⟩; simp_rw dist_eq_norm,
+  calc _ = ∥(a • x - a • x) + b • (x - z)∥
+    : by rw [smul_sub, sub_add_sub_comm, convex.combo_self hab]
+  ... = _ : by rw [sub_self, zero_add, norm_smul_of_nonneg hb],
+  calc _ = ∥a • (x - z) + (b • z - b • z)∥
+    : by rw [smul_sub, sub_add_sub_comm, convex.combo_self hab]
+  ... = _ : by rw [sub_self, add_zero, norm_smul_of_nonneg ha],
+end
```

Comment:
```suggestion
lemma exists_dist_eq (x z : E) {a b : ℝ} (ha : 0 ≤ a) (hb : 0 ≤ b) (hab : a + b = 1) :
  ∃ y, dist x y = b * dist x z ∧ dist y z = a * dist x z :=
begin
  use a • x + b • z,
  nth_rewrite 0 [←one_smul ℝ x],
  nth_rewrite 3 [←one_smul ℝ z],
  simp [dist_eq_norm, ←hab, add_smul, ←smul_sub, norm_smul_of_nonneg, ha, hb],
end
```
Also works, and is shorter. If you don't like the simp for whatever reason, then you should add braces to show that different goals are solved in each calc.